sources/Models/Vector3.hpp

```diff
@@ -136,6 +136,12 @@ namespace BBM
 		{
 			double mag = this->magnitude();
```

Comment:
this function call can be really costly so call only when need is certain

Reply:
What do you mean? How should I divide by the magnitude if I don't have it? Or do you want me to optimise a 0 ** 2 call? I don't believe that is necesarry.

Reply:
yeah